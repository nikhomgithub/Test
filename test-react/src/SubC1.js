import React, { Component } from 'react'

export default class SubC1 extends Component {
    render() {
        const {album}=this.props
        return (
            <li>
                <h3>{album.id}</h3>
                
            </li>
        )
    }
}
