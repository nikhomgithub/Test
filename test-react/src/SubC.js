import React, { Component } from 'react'
import SubC1 from './SubC1'
export default class SubC extends Component {
    render() {
        const {albums}=this.props
        return (
            <div>
                <ul>
                    {albums.map(album=>
                        <SubC1 key={album.id} album={album}></SubC1>
                    )}
                </ul>
            </div>
        )
    }
}
