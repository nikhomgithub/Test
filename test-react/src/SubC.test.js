import React from 'react';
import SubC1 from './SubC1';
import SubC from './SubC';
import { shallow , mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer' 

Enzyme.configure({ adapter: new Adapter() });


describe('This is Test',()=>{
  //Inject album to component as props
  
  //To find word 'An Album' in h1.text or in page
  it('1 rendes albums',()=>{

    const albums=[
        { id:1, title:"Jazz"},
        { id:2, title:"Rock"}
    ]
      
    const wrapper=shallow(<SubC albums={albums}/>)
    
    albums.forEach(album=>{
      expect(
        wrapper.contains(<SubC1 album={album}/>)
      ).toBe(true)
    })//forEach
  })

  
  it('2 renderer',()=>{

    const albums=[
        { id:1, title:"Jazz"},
        { id:2, title:"Rock"}
    ]
      
    const output=renderer.create(<SubC albums={albums}/>)
    
    expect(output).toMatchSnapshot()
  })
  

})