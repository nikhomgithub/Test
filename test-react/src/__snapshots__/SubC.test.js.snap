// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is Test 2 renderer 1`] = `
<div>
  <ul>
    <li>
      <h3>
        1
      </h3>
      <h4>
        Jazz
      </h4>
    </li>
    <li>
      <h3>
        2
      </h3>
      <h4>
        Rock
      </h4>
    </li>
  </ul>
</div>
`;
